File: algorithms/1991.find-the-middle-index-in-array/solution.rs
impl Solution {
    pub fn find_middle_index(nums: Vec<i32>) -> i32 {
        let sum = nums.iter().sum::<i32>();
        let (mut left, mut right) = (0, sum);

        for i in 0..nums.len() {
            right -= nums[i];

            if left == right {
                return i as i32;
            }

            left += nums[i];
        }

        -1
    }
}

File: algorithms/1991.find-the-middle-index-in-array/solution.cpp
#include <vector>

using namespace std;

class Solution {
  public:
    int findMiddleIndex(vector<int>& nums) {
        int sum = 0;
        for (int i = 0; i < nums.size(); i++) {
            sum += nums[i];
        }

        int left = 0;
        int right = sum;
        for (int i = 0; i < nums.size(); i++) {
            right -= nums[i];

            if (left == right) {
                return i;
            }

            left += nums[i];
        }

        return -1;
    }
};
